### PR TITLE
Update: Dimension filter uses primaryKey tag

### PR DESCRIPTION
**Pull request by cythrawll**
Dimensions don't always use `id` as their primary key.  We have tags set up to specify which dimension field is the primary key, and therefore we can build the correct dimension filter that gives the intended results.

### CODE REVIEW

File: packages/data/addon/mirage/routes/bard-lite.js [hunk 1]
```diff
@@ -72,10 +72,14 @@ function _fakeDimensionValues(name, count) {
   let fakeValues = [];
 
   for (let i = 0; i < count; i++) {
-    fakeValues.push({
+    let key = null;
+    if(name === 'multiSystemId') {
```
Comment (balajimanoharan) [inline packages/data/addon/mirage/routes/bard-lite.js]: Can we add some comments around this?

File: packages/reports/addon/consumers/request/filter.js [hunk 2]
```diff
@@ -73,9 +73,9 @@ export default ActionConsumer.extend({
      * @param {Object} route - route that has a model that contains a request property
      * @param {Object} dimension - dimension to filter
      */
-    [RequestActions.ADD_DIM_FILTER]: ({ currentModel }, dimension) => {
+    [RequestActions.ADD_DIM_FILTER]: ({ currentModel }, dimension)   => {
       get(currentModel, 'request').addFilter(
-        assign({ dimension }, DEFAULT_DIM_FILTER)
+        assign({ dimension}, DEFAULT_DIM_FILTER, {field: get(dimension, 'primaryKeyFieldName') || 'id'})
```
Comment (balajimanoharan) [inline packages/reports/addon/consumers/request/filter.js]: `primaryKeyFieldName` already defaults to `id`.
https://github.com/yahoo/navi/blob/fc6bb19c3b376c9309f49911aa1f79c6291e3a65/packages/data/addon/models/metadata/dimension.js#L87

File: packages/reports/addon/consumers/request/filter.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ export default ActionConsumer.extend({
   /**
    * @property {Ember.Service} requestActionDispatcher
    */
-  requestActionDispatcher: inject.service(),
+  requestActionDispatcher: inject.service(),
```
Comment (balajimanoharan) [inline packages/reports/addon/consumers/request/filter.js]: Can we revert this change?

File: packages/visualizations/addon/templates/components/cell-renderers/dimension.hbs [hunk 4]
```diff
@@ -1,6 +1,10 @@
 {{!-- Copyright 2017, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
 {{#with (get data (concat column.field.dimension '|desc' )) as |description|}}
-    <span title={{description}}>{{description}}</span>
+    {{#with (get data (concat column.field.dimension '|id' )) as |id|}}
+        <span title={{id}}>{{description}}</span>
```
Comment (balajimanoharan) [inline packages/visualizations/addon/templates/components/cell-renderers/dimension.hbs]: Can we do `{{description}} ({{id}})` in the title? I think we follow the same format elsewhere.

File: packages/visualizations/addon/templates/components/cell-renderers/dimension.hbs [hunk 5]
```diff
@@ -1,6 +1,10 @@
 {{!-- Copyright 2017, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
 {{#with (get data (concat column.field.dimension '|desc' )) as |description|}}
-    <span title={{description}}>{{description}}</span>
+    {{#with (get data (concat column.field.dimension '|id' )) as |id|}}
```
Comment (balajimanoharan) [inline packages/visualizations/addon/templates/components/cell-renderers/dimension.hbs]: This is done on line #9 as well. Can we make this the outer `with`?

File: packages/reports/addon/consumers/request/filter.js [hunk 6]
```diff
@@ -73,9 +73,9 @@ export default ActionConsumer.extend({
      * @param {Object} route - route that has a model that contains a request property
      * @param {Object} dimension - dimension to filter
      */
-    [RequestActions.ADD_DIM_FILTER]: ({ currentModel }, dimension) => {
+    [RequestActions.ADD_DIM_FILTER]: ({ currentModel }, dimension)   => {
```
Comment (jkusa) [inline packages/reports/addon/consumers/request/filter.js]: Can we revert this change?

File: packages/reports/addon/consumers/request/filter.js (hunk 2 above)
Comment (jkusa) [reply packages/reports/addon/consumers/request/filter.js]: Does it make sense to have `DEFAULT_DIM_FILTER` now?

Comment (cythrawll) [reply packages/reports/addon/consumers/request/filter.js]: So doing that assumes that the consumer is a dimension object, and in practice it seems that it is from where I tested it. However, a lot of tests pass in just the dimension name.  Is this bad mocks? or will we actually run into dimension simply being a string?

File: packages/visualizations/addon/templates/components/cell-renderers/dimension.hbs (hunk 4 above)
Comment (cythrawll) [reply packages/visualizations/addon/templates/components/cell-renderers/dimension.hbs]: we do, but only when desc and ID are far from each other.  this isn't the case here.

File: packages/visualizations/addon/templates/components/cell-renderers/dimension.hbs (hunk 5 above)
Comment (cythrawll) [reply packages/visualizations/addon/templates/components/cell-renderers/dimension.hbs]: would that actually fix conditions for all 4 states?

File: packages/reports/addon/consumers/request/filter.js (hunk 2 above)
Comment (jkusa) [reply packages/reports/addon/consumers/request/filter.js]: Bad mocks I hope. I might be time to consider propTypes or flow 😬

File: packages/visualizations/addon/templates/components/cell-renderers/dimension.hbs (hunk 5 above)
Comment (cythrawll) [reply packages/visualizations/addon/templates/components/cell-renderers/dimension.hbs]: gonna refactor this so that the component makes this decision rather than the template.

File: packages/visualizations/addon/components/cell-renderers/dimension.js [hunk 7]
```diff
@@ -10,14 +10,63 @@
...
+   * @property {String} idField - field to find id
+   */
+  idField: computed('column.field.dimension', function() {
+    return get(this, 'column.field.dimension') + '|id';
+  }),
+
+  /**
+   * @property {String} title - value that should be used in hoverover title
+   */
+  title: computed('descField', 'idField', 'data', function() {
```
Comment (balajimanoharan) [inline packages/visualizations/addon/components/cell-renderers/dimension.js]: screen shot 2018-08-10 at 10 23 55 am

I wasn't sure what you meant by that. But I will explain what I meant.
In the table, unless you choose an column that displays the `id`, the `id` is not seen when a description is available. When we talked to some of the spotlight users, they were more familiar with the `id` than the `desc`. So on hover, if we can show both, that would be an improvement. `description (id)`.

File: packages/visualizations/addon/components/cell-renderers/dimension.js [hunk 8]
```diff
@@ -10,14 +10,63 @@
...
+        idField = get(this, 'idField'),
+        data = get(this, 'data');
+
+    if(data[idField]) {
+      return data[idField];
+    } else if(data[descField]) {
+      return data[descField];
+    }
+
+    return '--';
```
Comment (balajimanoharan) [inline packages/visualizations/addon/components/cell-renderers/dimension.js]: Do we need a `title` in this case?

File: packages/visualizations/addon/components/cell-renderers/dimension.js [hunk 9]
```diff
@@ -10,14 +10,63 @@
...
+  }),
+
+  /**
+   * @property {String} value - value that should be displayed in table cell
+   */
+  value: computed('descField', 'idField', 'data', function() {
```
Comment (balajimanoharan) [inline packages/visualizations/addon/components/cell-renderers/dimension.js]: Is this the exact same as `title`?

Comment (balajimanoharan) [reply packages/visualizations/addon/components/cell-renderers/dimension.js]: Do we already have tests for these changes?

File: packages/visualizations/addon/components/cell-renderers/dimension.js (hunk 8 above)
Comment (cythrawll) [reply packages/visualizations/addon/components/cell-renderers/dimension.js]: fixed

File: packages/visualizations/addon/components/cell-renderers/dimension.js (hunk 7 above)
Comment (cythrawll) [reply packages/visualizations/addon/components/cell-renderers/dimension.js]: I was just saying desc is right next to the id in this case during hover over, but I changed this behavior. to be `desc (id)` if both are present.

File: packages/visualizations/addon/components/cell-renderers/dimension.js (hunk 9 above)
Comment (cythrawll) [reply packages/visualizations/addon/components/cell-renderers/dimension.js]: display priority is different.

Tests added for titles
